### ISSUE TITLE
How to define hyper parameters for the fine-tuning task?

### ISSUE THREAD

**Issue by avdebut** [anchen1011/chatgpt-finetune-ui#3]
Currently the UI allows fine-tuning ChatGPT model but there is no way to define hyper parameters like number of epochs, batch size etc. It would be useful to have some configurable options to define these hyper parameters for the fine-tuning task running in the backend.

This will give more control and flexibility to the users to fine-tune the model as per their need by tuning these important parameters.

Some suggested parameters that can be configurable:

- Number of epochs
- Batch size
- Learning rate
- Optimizer (Adam, SGD etc)

Please let me know if adding these configurable options via the UI makes sense or if you have any other suggestions.

**Comment by anchen1011** [comment]
WIP!

**Comment by qaixerabbas** [comment]
@avdebut I don't think these parameters are available for finetuning (I am not 100% sure). [This Link](https://platform.openai.com/docs/guides/fine-tuning/analyzing-your-fine-tuned-model) has some information about analysis of finetuned models and there is not information about the parameters that you asked

Here is the object structure from the same link I shared

```
{
    "object": "fine_tuning.job.event",
    "id": "ftevent-abc-123",
    "created_at": 1693582679,
    "level": "info",
    "message": "Step 100/100: training loss=0.00",
    "data": {
        "step": 100,
        "train_loss": 1.805623287509661e-5,
        "train_mean_token_accuracy": 1.0
    },
    "type": "metrics"
}
```

**Comment by anchen1011** [comment]
Thanks! @qaixerabbas 

We have number of epochs supported now. I don't think other setups could be configured with OpenAI fine-tune API..
